packages/amplify_auth_cognito/android/src/main/kotlin/com/amazonaws/amplify/amplify_auth_cognito/types/FlutterInvalidStateException.kt

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazonaws.amplify.amplify_auth_cognito.types
+
+import com.amplifyframework.auth.AuthException
+
+class FlutterInvalidStateException: AuthException {
+    constructor(message: String, recoverySuggestion: String) : super(message, Exception("invalidSate"), recoverySuggestion)
+}
```

Comment:
Is an `InvalidStateException` something that can go into `Core` in the future? Or is this one `Auth` specific in iOS?

Reply:
To my knowledge [this is only used ](https://github.com/aws-amplify/amplify-ios/search?q=.invalidState)in amplify-ios Auth.